src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4896,18 +4896,15 @@ static QVariant executeGeomOverlay( const QVariantList &values, const QgsExpress
   QVariant layerValue = node->eval( parent, context );
   ENSURE_NO_EVAL_ERROR
 
-  int paramIndex = 1;
-
   QString subExpString;
   QgsExpression subExpression;
-  if ( !testOnly )
-  {
-    // Second parameter is the subexpression (invalid QVariant for boolean)
-    node = QgsExpressionUtils::getNode( values.at( paramIndex ), parent );
-    ENSURE_NO_EVAL_ERROR
-
-    subExpString = node->dump();
-    ++ paramIndex;
+  if ( values.length() > 1 ){
```

Comment:
There will always be the maximum number of parameters (currently 3). If the user does not specify a parameter, it will have the default value of the parameter. So this check is not required.